clients/java/src/main/java/io/camunda/client/CamundaClient.java

```diff
@@ -0,0 +1,588 @@
+   *  .newDeployCommand()
+   *  .addResourceFile("~/wf/process1.bpmn")
+   *  .addResourceFile("~/wf/process2.bpmn")
+   *  .addResourceFile("~/dmn/decision.dmn")
+   *  .send();
+   * </pre>
+   *
+   * @return a builder for the command
+   */
+  DeployResourceCommandStep1 newDeployResourceCommand();
+
+  /**
+   * Command to create/start a new instance of a process.
+   *
+   * <pre>
+   * zeebeClient
+   *  .newCreateInstanceCommand()
+   *  .bpmnProcessId("my-process")
+   *  .latestVersion()
+   *  .variables(json)
+   *  .send();
+   * </pre>
+   *
+   * @return a builder for the command
+   */
+  CreateProcessInstanceCommandStep1 newCreateInstanceCommand();
+
+  /**
+   * Command to modify a process instance.
+   *
+   * <pre>
+   *   zeebeClient
+   *    .newModifyProcessInstanceCommand(processInstanceKey)
+   *    .activateElement("element1")
+   *    .and()
+   *    .activateElement("element2")
+   *    .withVariables(globalScopedVariables)
+   *    .withVariables(localScopedVariables, "element2")
+   *    .and()
+   *    .terminateElement("element3")
+   *    .send();
+   * </pre>
+   *
+   * @param processInstanceKey the key which identifies the corresponding process instance
+   * @return a builder for the command
+   */
+  ModifyProcessInstanceCommandStep1 newModifyProcessInstanceCommand(long processInstanceKey);
+
+  /**
+   * Command to migrate a process instance to a different process definition.
+   *
+   * <p>The migration command contains a migration plan. Migration plan contains
+   * targetProcessDefinitionKey to indicate which process definition to use for the migration.
+   * Mapping instructions for the migration describe how to map elements from the source process
+   * definition to the target process definition.
+   *
+   * <p>For example, let's consider we want to migrate process instance with key {@code 1}, target
+   * process definition key {@code 2}, a source process definition with a service task with id
+   * {@code "task1"} and the target process definition with a service task with id {@code "task2"}.
+   * The migration command could be:
+   *
+   * <pre>{@code
+   * {
+   *  "processInstanceKey": 1,
+   *  "migrationPlan": {
+   *   "targetProcessDefinitionKey": 2,
+   *   "mappingInstructions": [
+   *    {
+   *     "sourceElementId": "task1",
+   *     "targetElementId": "task2"
+   *    }
+   *   ]
+   *  }
+   * }
+   * }</pre>
+   *
+   * <pre>
+   *
+   * zeebeClient
+   *  .newMigrateProcessInstanceCommand(1L)
+   *  .migrationPlan(2L)
+   *  .addMappingInstruction("element1", "element2")
+   *  .addMappingInstruction("element3", "element4")
+   *  .send();
+   * </pre>
+   *
+   * <pre>
+   * final MigrationPlan migrationPlan =
+   *         MigrationPlan.newBuilder()
+   *             .withTargetProcessDefinitionKey(2L)
+   *             .addMappingInstruction("element1", "element2")
+   *             .addMappingInstruction("element3", "element4")
+   *             .build();
+   * zeebeClient
```

Comment:
🔧 There are still a lot of those examples referring to `zeebeClient`. Not a showstopper, of course, it's just documentation.